### PR TITLE
Add options to pagelist

### PR DESCRIPTION
**Pull request by chang-zhao**
So that limiting search depth, hash calculation and skipping ACL could be requested.
#11

### CODE REVIEW

File: dokuwikixmlrpc.py [hunk 1]
```diff
@@ -247,9 +247,9 @@ def append_page(self, page_id, text, summary='', minor=False):
         self._xmlrpc.dokuwiki.appendPage(page_id, text, params)
 
     @checkerr
-    def pagelist(self, namespace):
+    def pagelist(self, namespace, opts = {'depth': 0, 'hash': False, 'skipacl': False}):
```
Comment (kynan) [inline dokuwikixmlrpc.py:250]: What do these options do? I can't really find any documentation for this.

Comment (chang-zhao) [reply dokuwikixmlrpc.py:250]: Here: https://www.dokuwiki.org/devel:xmlrpc#dokuwikigetpagelist

There's a link to XRef documentation on [search_allpages()](http://xref.dokuwiki.org/reference/dokuwiki/_functions/search_allpages.html)

where it says:

```
$opts['depth']   recursion level, 0 for all
$opts['hash']    do md5 sum of content?
$opts['skipacl'] list everything regardless of ACL
```

File: dokuwikixmlrpc.py [hunk 2]
```diff
@@ -247,9 +247,9 @@ def append_page(self, page_id, text, summary='', minor=False):
...
         """Lists all pages within a given namespace."""
```
Comment (kynan) [inline dokuwikixmlrpc.py:251]:
```suggestion
        """Lists all pages within a given namespace.
        
        :param str namespace: The namespace to list pages for
        :param dict opts: Options (optional)
        * :depth: recursion level, default to 0, for all
        * :hash: if True, do the md5 sum of the content, defaults to False
        * :skipacl: if True, list everything regardless of ACL, defaults to False
        """
```

**Review by kynan — APPROVED** [review]
Thanks. I've added this as documentation.

**Pull request merged**